tools/scripts/build_projects.py

```diff
@@ -81,6 +81,25 @@ def shell_source(script):
 
 	os.environ.update(env)
 
+def re_run_stm32(cmd):
+	global ERR
+	log("Project first failed, rebuild to check if the error persists")
```

Comment:
nit: instead of Project first failed maybe rewrite to First build of the project failed, rebuild ...